### DISCUSSION TITLE
Multiple coding languages

### DISCUSSION THREAD

**Discussion by fxvits** [zoriya/Kyoo#631] [General]
Just by curiosity why did you use multiple coding languages in your solution ? is it a personal challenge, for fun :) ?
Transcoder => go
Scanner => python
Backend => C#

**Comment by zoriya** [comment]
There is no benefit in doing everything in the same language. Spliting the code in multiple services make writing & maintaining code easier. It also is better for scalability.

To give insights of why specific languages where used, it's pretty simple:
transcoder -> simple http server that wraps ffmpeg & sync state concurrently -> go is good for concurrent stuff & api

scanner -> do http requests from online databases, map it to kyoo's format and send it -> python or js would be the bests languages for that, i just picked one

backend -> an api, any api service could work. I picked c# 5 years ago because it's good for api and i was a bit familiar with it. This will be changed to js for the v5 (see #597 for context)
